packages/@expo/cli/src/start/server/metro/withMetroMultiPlatform.ts

```diff
@@ -481,7 +484,9 @@ export async function withMetroMultiPlatformAsync(
   await setupNodeExternals(projectRoot);
 
   let expoConfigPlatforms = Object.entries(platformBundlers)
-    .filter(([, bundler]) => bundler === 'metro')
+    .filter(
+      ([platform, bundler]) => bundler === 'metro' && exp.platforms?.includes(platform as Platform)
```

Comment:
This new guard is required as `expoConfigPlatforms` included `web` for non-web projects.